### ISSUE TITLE
Entities: Make `InitializeState` support Async operations

### ISSUE THREAD

**Issue by gorillapower** [microsoft/durabletask-dotnet#263]
I have a requirement to create initial Entity state from an external source (db, blob, etc). 
However currently, the override to create initial state for an Entity does not support async operations, forcing me to do syncronous calls.

*Suggestion*
```csharp
protected override async Task<T> InitializeStateAsync(TaskEntityOperation operation)
{
     //perform async operations
}

````

**Comment by jhinm112** [comment]
Thread is also blocked in my initializeState because async operation is not supported

**Comment by gorillapower** [comment]
Thanks @jviau. Would this pattern still work with this method signature?
``` csharp
protected override async Task<T> InitializeStateAsync(TaskEntityOperation operation)
{
     //perform async operations
}

```
as I need to return `T` not void (as the entity inherites from `TaskEntity<T>`) and it looks like the base class will store the returned value in the `base.State` property.

Full Code Example
``` csharp
public class T : TaskEntity<T>
{
    public void MyMethod() { }
    
    [Function(nameof(T))]
    public static Task RunEntityAsync([EntityTrigger] TaskEntityDispatcher dispatcher)
    {
        return dispatcher.DispatchAsync<WorkflowStateDurableEntity>();
    }
    
    protected override T InitializeState(TaskEntityOperation operation)
    {
        //inject EnsureInitializeAsync pattern here?
        T state = new();
        return state; 
    }
}
```

Also worth noting, an additional alternative would be to use the[ functions based syntax](https://learn.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-dotnet-entities?pivots=isolated#function-based-syntax) which would allow the populating of the state inside the `DispatchAsync()`

**Comment by jviau** [comment]
@gorillapower

Yes, returning `default(T)` would be fine there. `InitializeState` is just a convenience method for populating state when there is none. I updated my sample to show that.

You could also set it in `InitializeState` and then populate its contents in `InitializeAsync`.

**Comment by YunchuWang** [comment]
workaround provided, supporting async initstate requires major entity architecture changes not planned atm
